app/src/main/java/com/esri/arcgisruntime/opensourceapps/datacollection/views/popup/PopupView.kt

```diff
@@ -44,9 +60,36 @@ class PopupView : FrameLayout {
     var popup: Popup? = null
         set(value) {
             field = value
-            popupAttributeListAdapter.submitList(getPopupAttributeList())
+            popupAttributeListAdapter.submitList(popupManager.displayedFields)
         }
 
+    var editMode: Boolean = false
+        set(value) {
+            field = value
+            if (field) {
+                popupAttributeListAdapter.submitList(popupManager.editableFields)
+                popupManager.startEditing()
+            } else if (popup != null) {
+                popupAttributeListAdapter.submitList(popupManager.displayedFields)
+                if (!saveEdit) {
+                    popupManager.cancelEditing()
+                }
+            }
+            popupAttributeListAdapter.notifyDataSetChanged()
+        }
+
+    var saveEdit: Boolean = false
+        set(value) {field = value
```

Comment:
This could go in a new line 
```suggestion
        set(value) {
        field = value
```

Reply:
No longer relevant